app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/posts/EventPost.kt

```diff
@@ -173,12 +182,67 @@ fun EventPost(
               }
             }
 
+        if (currentUser == post.userId) {
+          IconButton(onClick = { showDeletePostDialog = true }) {
+            Icon(
+                imageVector = Icons.Default.Delete,
+                contentDescription = "Delete",
+                tint = MaterialTheme.colorScheme.onBackground)
+          }
+        }
+
         Spacer(modifier = Modifier.weight(1f))
         Text(
-            text = "${event.getDateString()}, ${event.getTimeString()}",
+            text = "${event.date}, ${event.time}",
             style = MaterialTheme.typography.bodyMedium,
             color = Grey)
       }
     }
   }
+
+  if (showDeletePostDialog) {
+    DeletePostDialog(
+        onConfirm = {
+          coroutineScope.launch {
```

Comment:
Now this is what I like to see. Clean, concise and every function is in the right class (according to me in any case)!